src/xopen/__init__.py

```diff
@@ -196,12 +196,20 @@ def __init__(self, path, mode='r', threads=None):
         """
         if mode not in ('r', 'rt', 'rb'):
             raise ValueError("Mode is '{0}', but it must be 'r', 'rt' or 'rb'".format(mode))
-        if threads is None:
-            threads = min(_available_cpu_count(), 4)
-        
+
         pigz_args = ['pigz', '-cd', path]
-        if threads != 0:
-            pigz_args += ['-p', str(threads)]
+
+        if not threads:
```

Comment:
Hm, now I would actually write it as `threads is None` because the purpose is to explicitly check whether the user set something different than the default `None`, not to check the falsiness. Perhaps I’m being inconsistent here, but since you found the more explicit version more readable anyway, can you change it? Sorry for the back and forth.

Reply:
Since it’s me who’s to blame here, I’ll just merge this and fix it up afterwards.

Reply:
Well it actually was `if threads is None or threads == 0:`
But then I thought I was droning on again about my style preference.

`0` is a weird case anyway. Maybe that should be reserved for running the pigz default?